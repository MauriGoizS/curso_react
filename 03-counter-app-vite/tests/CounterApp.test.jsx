const { render, screen, fireEvent } = require("@testing-library/react")
const { CounterApp } = require("../src/CounterApp")

describe('Test in <CounterApp /> ', () => {

    const initialValue = 100;

    test('Must match the snapshot', () => {

        const { container } = render( <CounterApp value={ initialValue }/> )
        expect( container ).toMatchSnapshot();

    });
    
    test('Should show the initial value of 100 <CounterApp value={ 100 }/>', () => {
        
        render(<CounterApp value={ initialValue }/>);        
        expect(screen.getByText("100")).toBeTruthy();

    })
    
    test('Must be increased with the +1 button', () => {
    
        render(<CounterApp value={ initialValue }/>);        
        fireEvent.click( screen.getByText('+1') );
        expect( screen.getByText('101') ).toBeTruthy();

    });

    test('Must be decremented with the -1 button', () => {
    
        render(<CounterApp value={ initialValue }/>);        
        fireEvent.click( screen.getByText('-1') );
        expect( screen.getByText('99') ).toBeTruthy();

    });

    test('The reset button should work.', () => {
        
        render(<CounterApp value={ 355 }/>); 
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        // fireEvent.click( screen.getByText('reset') );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-reset' } ) );

        // expect( screen.getByText( 355 ) ).toBeTruthy();

    })
    
    

});
