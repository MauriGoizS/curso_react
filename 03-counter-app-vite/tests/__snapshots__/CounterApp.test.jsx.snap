// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CounterApp />  Must match the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
  </h2>
  <button>
     -1 
  </button>
  <button>
     reset 
  </button>
  <button
    aria-label="btn-reset"
  >
     +1 
  </button>
</div>
`;
